app/components/geoblacklight/arc_gis_component.rb

```diff
@@ -0,0 +1,25 @@
+# frozen_string_literal: true
+
+module Geoblacklight
+  class ArcGisComponent < ViewComponent::Base
+    attr_reader :document
+
+    def initialize(document:, action:, **)
+      @document = document
+      super
+    end
+
+    # Generates an ArcGIS.com viewer url with params that can open content directly
+    # See: https://doc.arcgis.com/en/arcgis-online/reference/use-url-parameters.htm
+    def arcgis_link(arcgis_urls)
```

Comment:
I don't think this is a necessary change but you could update this to just have the argis_urls to be document.arcgis_urls

Reply:
right, yes -- i just removed the need to pass in the urls and just pulled them straight from the document which is available to the component class (and updated the specs)